### DISCUSSION TITLE
Disable 2049 port forwarding?

### DISCUSSION THREAD

**Discussion by qtcwt** [lima-vm/lima#2199] [Q&A]
Hi community! I would like to disable 2049 port forwarding so that in the guest VM I can do development that uses 2049 port by default. Is there a way to do so? 

I tried to ignore 2049 forwarding by doing the following in default/lima.yaml:
```
portForwards:
  - guestPort: 2049
    ignore: true
```

But with this configuration, during boot, port 2049 is still forwarded.

My host machine is MacBook M3 Pro, macOS Sonoma 14.3. My guest VM is aarch64 Ubuntu. `vmType: vz`. I mounted one writeable directory in addition to the default read-only home directory. The mount type is virtiofs (as default for vz). forwardX11 enabled. Rosetta enabled.

`sudo ss -lptn 'sport = :2049'` reports that 2049 is actively listened, but the process name is empty:
```bash
$sudo ss -lptn 'sport = :2049'
State            Recv-Q           Send-Q                       Local Address:Port                       Peer Address:Port           Process
LISTEN           0                64                                 0.0.0.0:2049                            0.0.0.0:*
LISTEN           0                64                                    [::]:2049                               [::]:*
```

Thanks!

**Comment by jandubois** [comment]
I don't understand the question. How does it "not work"?

Even if you do forward the port, it should still be usable inside the VM unless something else is listening on the port.

**Reply by qtcwt** [reply]
Not working means that after I configured forwardPorts as I described, the port 2049 was still forwarded. The related log entries during boot:
```
INFO[0011] [hostagent] Forwarding TCP from 0.0.0.0:2049 to 127.0.0.1:2049
INFO[0011] [hostagent] Forwarding TCP from [::]:2049 to 127.0.0.1:2049
WARN[0011] [hostagent] failed to set up forwarding tcp port 2049 (negligible if already forwarded)
```

`sudo ss -lptn 'sport = :2049'` reports that 2049 is actively listened, but the process name is empty:
```bash
sudo ss -lptn 'sport = :2049'
State            Recv-Q           Send-Q                       Local Address:Port                       Peer Address:Port           Process
LISTEN           0                64                                 0.0.0.0:2049                            0.0.0.0:*
LISTEN           0                64                                    [::]:2049                               [::]:*
```

**Comment by afbjorklund** [comment]
I think there is a typo there, in `portForwards`

`error ... unknown field \"portForward\" ...`

**Reply by qtcwt** [reply]
Oops. I manually typed what I configured in the question, and yes that was a typo. In my lima.yaml it is `portForwards`.

**Comment by kbrock** [comment]
can this be marked as answered?

**Reply by qtcwt** [reply]
Well I haven't received an answer that fixes my issue.
